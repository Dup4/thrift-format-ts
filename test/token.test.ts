import { describe, expect, it } from "vitest";

import { ThriftData } from "thrift-parser-typescript";
import { readFile } from "./common";

describe("Thrift Parser", () => {
  it("view token", () => {
    const content = readFile("token.thrift");
    const thriftData = ThriftData.fromString(content);
    const tokens = thriftData.tokenStream.getTokens();

    for (const token of tokens) {
      expect(
        `[line=${token.line}] [type=${token.type}] [channel=${token.channel}] [tokenIndex=${token.tokenIndex}] [startIndex=${token.startIndex}] [stopIndex=${token.stopIndex}]\n${token.text}`,
      ).matchSnapshot();
    }
  });
});
