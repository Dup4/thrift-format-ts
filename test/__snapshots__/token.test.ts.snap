// Vitest Snapshot v1

exports[`Thrift Parser > view token 1`] = `
"[line=1] [type=52] [channel=2] [tokenIndex=0] [startIndex=0] [stopIndex=21]
// ServiceLineComment
"
`;

exports[`Thrift Parser > view token 2`] = `
"[line=2] [type=51] [channel=1] [tokenIndex=1] [startIndex=22] [stopIndex=22]

"
`;

exports[`Thrift Parser > view token 3`] = `
"[line=3] [type=17] [channel=0] [tokenIndex=2] [startIndex=23] [stopIndex=29]
service"
`;

exports[`Thrift Parser > view token 4`] = `
"[line=3] [type=51] [channel=1] [tokenIndex=3] [startIndex=30] [stopIndex=30]
 "
`;

exports[`Thrift Parser > view token 5`] = `
"[line=3] [type=49] [channel=0] [tokenIndex=4] [startIndex=31] [stopIndex=36]
ApiTag"
`;

exports[`Thrift Parser > view token 6`] = `
"[line=3] [type=51] [channel=1] [tokenIndex=5] [startIndex=37] [stopIndex=37]
 "
`;

exports[`Thrift Parser > view token 7`] = `
"[line=3] [type=11] [channel=0] [tokenIndex=6] [startIndex=38] [stopIndex=38]
{"
`;

exports[`Thrift Parser > view token 8`] = `
"[line=3] [type=51] [channel=1] [tokenIndex=7] [startIndex=39] [stopIndex=42]

   "
`;

exports[`Thrift Parser > view token 9`] = `
"[line=4] [type=52] [channel=2] [tokenIndex=8] [startIndex=43] [stopIndex=57]
// Bar Comment
"
`;

exports[`Thrift Parser > view token 10`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=9] [startIndex=58] [stopIndex=60]
   "
`;

exports[`Thrift Parser > view token 11`] = `
"[line=5] [type=26] [channel=0] [tokenIndex=10] [startIndex=61] [stopIndex=64]
void"
`;

exports[`Thrift Parser > view token 12`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=11] [startIndex=65] [stopIndex=65]
 "
`;

exports[`Thrift Parser > view token 13`] = `
"[line=5] [type=49] [channel=0] [tokenIndex=12] [startIndex=66] [stopIndex=68]
Bar"
`;

exports[`Thrift Parser > view token 14`] = `
"[line=5] [type=22] [channel=0] [tokenIndex=13] [startIndex=69] [stopIndex=69]
("
`;

exports[`Thrift Parser > view token 15`] = `
"[line=5] [type=23] [channel=0] [tokenIndex=14] [startIndex=70] [stopIndex=70]
)"
`;

exports[`Thrift Parser > view token 16`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=15] [startIndex=71] [stopIndex=71]
 "
`;

exports[`Thrift Parser > view token 17`] = `
"[line=5] [type=22] [channel=0] [tokenIndex=16] [startIndex=72] [stopIndex=72]
("
`;

exports[`Thrift Parser > view token 18`] = `
"[line=5] [type=49] [channel=0] [tokenIndex=17] [startIndex=73] [stopIndex=80]
api.post"
`;

exports[`Thrift Parser > view token 19`] = `
"[line=5] [type=8] [channel=0] [tokenIndex=18] [startIndex=81] [stopIndex=81]
="
`;

exports[`Thrift Parser > view token 20`] = `
"[line=5] [type=48] [channel=0] [tokenIndex=19] [startIndex=82] [stopIndex=87]
\\"/Bar\\""
`;

exports[`Thrift Parser > view token 21`] = `
"[line=5] [type=50] [channel=0] [tokenIndex=20] [startIndex=88] [stopIndex=88]
,"
`;

exports[`Thrift Parser > view token 22`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=21] [startIndex=89] [stopIndex=89]
 "
`;

exports[`Thrift Parser > view token 23`] = `
"[line=5] [type=49] [channel=0] [tokenIndex=22] [startIndex=90] [stopIndex=104]
api.data_policy"
`;

exports[`Thrift Parser > view token 24`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=23] [startIndex=105] [stopIndex=105]
 "
`;

exports[`Thrift Parser > view token 25`] = `
"[line=5] [type=8] [channel=0] [tokenIndex=24] [startIndex=106] [stopIndex=106]
="
`;

exports[`Thrift Parser > view token 26`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=25] [startIndex=107] [stopIndex=107]
 "
`;

exports[`Thrift Parser > view token 27`] = `
"[line=5] [type=48] [channel=0] [tokenIndex=26] [startIndex=108] [stopIndex=115]
\\"ENABLE\\""
`;

exports[`Thrift Parser > view token 28`] = `
"[line=5] [type=50] [channel=0] [tokenIndex=27] [startIndex=116] [stopIndex=116]
,"
`;

exports[`Thrift Parser > view token 29`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=28] [startIndex=117] [stopIndex=117]
 "
`;

exports[`Thrift Parser > view token 30`] = `
"[line=5] [type=49] [channel=0] [tokenIndex=29] [startIndex=118] [stopIndex=131]
api.serializer"
`;

exports[`Thrift Parser > view token 31`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=30] [startIndex=132] [stopIndex=132]
 "
`;

exports[`Thrift Parser > view token 32`] = `
"[line=5] [type=8] [channel=0] [tokenIndex=31] [startIndex=133] [stopIndex=133]
="
`;

exports[`Thrift Parser > view token 33`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=32] [startIndex=134] [stopIndex=134]
 "
`;

exports[`Thrift Parser > view token 34`] = `
"[line=5] [type=48] [channel=0] [tokenIndex=33] [startIndex=135] [stopIndex=140]
\\"json\\""
`;

exports[`Thrift Parser > view token 35`] = `
"[line=5] [type=50] [channel=0] [tokenIndex=34] [startIndex=141] [stopIndex=141]
,"
`;

exports[`Thrift Parser > view token 36`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=35] [startIndex=142] [stopIndex=142]
 "
`;

exports[`Thrift Parser > view token 37`] = `
"[line=5] [type=49] [channel=0] [tokenIndex=36] [startIndex=143] [stopIndex=161]
api.resp_serializer"
`;

exports[`Thrift Parser > view token 38`] = `
"[line=5] [type=8] [channel=0] [tokenIndex=37] [startIndex=162] [stopIndex=162]
="
`;

exports[`Thrift Parser > view token 39`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=38] [startIndex=163] [stopIndex=163]
 "
`;

exports[`Thrift Parser > view token 40`] = `
"[line=5] [type=48] [channel=0] [tokenIndex=39] [startIndex=164] [stopIndex=169]
\\"json\\""
`;

exports[`Thrift Parser > view token 41`] = `
"[line=5] [type=23] [channel=0] [tokenIndex=40] [startIndex=170] [stopIndex=170]
)"
`;

exports[`Thrift Parser > view token 42`] = `
"[line=5] [type=50] [channel=0] [tokenIndex=41] [startIndex=171] [stopIndex=171]
,"
`;

exports[`Thrift Parser > view token 43`] = `
"[line=5] [type=51] [channel=1] [tokenIndex=42] [startIndex=172] [stopIndex=176]


   "
`;

exports[`Thrift Parser > view token 44`] = `
"[line=7] [type=52] [channel=2] [tokenIndex=43] [startIndex=177] [stopIndex=191]
// Baz Comment
"
`;

exports[`Thrift Parser > view token 45`] = `
"[line=8] [type=51] [channel=1] [tokenIndex=44] [startIndex=192] [stopIndex=200]

    
   "
`;

exports[`Thrift Parser > view token 46`] = `
"[line=10] [type=52] [channel=2] [tokenIndex=45] [startIndex=201] [stopIndex=215]
// Baz Comment
"
`;

exports[`Thrift Parser > view token 47`] = `
"[line=11] [type=51] [channel=1] [tokenIndex=46] [startIndex=216] [stopIndex=218]
   "
`;

exports[`Thrift Parser > view token 48`] = `
"[line=11] [type=52] [channel=2] [tokenIndex=47] [startIndex=219] [stopIndex=233]
// Baz Comment
"
`;

exports[`Thrift Parser > view token 49`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=48] [startIndex=234] [stopIndex=237]
    "
`;

exports[`Thrift Parser > view token 50`] = `
"[line=12] [type=26] [channel=0] [tokenIndex=49] [startIndex=238] [stopIndex=241]
void"
`;

exports[`Thrift Parser > view token 51`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=50] [startIndex=242] [stopIndex=242]
 "
`;

exports[`Thrift Parser > view token 52`] = `
"[line=12] [type=49] [channel=0] [tokenIndex=51] [startIndex=243] [stopIndex=245]
Baz"
`;

exports[`Thrift Parser > view token 53`] = `
"[line=12] [type=22] [channel=0] [tokenIndex=52] [startIndex=246] [stopIndex=246]
("
`;

exports[`Thrift Parser > view token 54`] = `
"[line=12] [type=23] [channel=0] [tokenIndex=53] [startIndex=247] [stopIndex=247]
)"
`;

exports[`Thrift Parser > view token 55`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=54] [startIndex=248] [stopIndex=248]
 "
`;

exports[`Thrift Parser > view token 56`] = `
"[line=12] [type=22] [channel=0] [tokenIndex=55] [startIndex=249] [stopIndex=249]
("
`;

exports[`Thrift Parser > view token 57`] = `
"[line=12] [type=49] [channel=0] [tokenIndex=56] [startIndex=250] [stopIndex=257]
api.post"
`;

exports[`Thrift Parser > view token 58`] = `
"[line=12] [type=8] [channel=0] [tokenIndex=57] [startIndex=258] [stopIndex=258]
="
`;

exports[`Thrift Parser > view token 59`] = `
"[line=12] [type=48] [channel=0] [tokenIndex=58] [startIndex=259] [stopIndex=264]
\\"/Baz\\""
`;

exports[`Thrift Parser > view token 60`] = `
"[line=12] [type=50] [channel=0] [tokenIndex=59] [startIndex=265] [stopIndex=265]
,"
`;

exports[`Thrift Parser > view token 61`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=60] [startIndex=266] [stopIndex=266]
 "
`;

exports[`Thrift Parser > view token 62`] = `
"[line=12] [type=49] [channel=0] [tokenIndex=61] [startIndex=267] [stopIndex=281]
api.data_policy"
`;

exports[`Thrift Parser > view token 63`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=62] [startIndex=282] [stopIndex=282]
 "
`;

exports[`Thrift Parser > view token 64`] = `
"[line=12] [type=8] [channel=0] [tokenIndex=63] [startIndex=283] [stopIndex=283]
="
`;

exports[`Thrift Parser > view token 65`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=64] [startIndex=284] [stopIndex=284]
 "
`;

exports[`Thrift Parser > view token 66`] = `
"[line=12] [type=48] [channel=0] [tokenIndex=65] [startIndex=285] [stopIndex=292]
\\"ENABLE\\""
`;

exports[`Thrift Parser > view token 67`] = `
"[line=12] [type=50] [channel=0] [tokenIndex=66] [startIndex=293] [stopIndex=293]
,"
`;

exports[`Thrift Parser > view token 68`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=67] [startIndex=294] [stopIndex=294]
 "
`;

exports[`Thrift Parser > view token 69`] = `
"[line=12] [type=49] [channel=0] [tokenIndex=68] [startIndex=295] [stopIndex=308]
api.serializer"
`;

exports[`Thrift Parser > view token 70`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=69] [startIndex=309] [stopIndex=309]
 "
`;

exports[`Thrift Parser > view token 71`] = `
"[line=12] [type=8] [channel=0] [tokenIndex=70] [startIndex=310] [stopIndex=310]
="
`;

exports[`Thrift Parser > view token 72`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=71] [startIndex=311] [stopIndex=311]
 "
`;

exports[`Thrift Parser > view token 73`] = `
"[line=12] [type=48] [channel=0] [tokenIndex=72] [startIndex=312] [stopIndex=317]
\\"json\\""
`;

exports[`Thrift Parser > view token 74`] = `
"[line=12] [type=50] [channel=0] [tokenIndex=73] [startIndex=318] [stopIndex=318]
,"
`;

exports[`Thrift Parser > view token 75`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=74] [startIndex=319] [stopIndex=319]
 "
`;

exports[`Thrift Parser > view token 76`] = `
"[line=12] [type=49] [channel=0] [tokenIndex=75] [startIndex=320] [stopIndex=338]
api.resp_serializer"
`;

exports[`Thrift Parser > view token 77`] = `
"[line=12] [type=8] [channel=0] [tokenIndex=76] [startIndex=339] [stopIndex=339]
="
`;

exports[`Thrift Parser > view token 78`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=77] [startIndex=340] [stopIndex=340]
 "
`;

exports[`Thrift Parser > view token 79`] = `
"[line=12] [type=48] [channel=0] [tokenIndex=78] [startIndex=341] [stopIndex=346]
\\"json\\""
`;

exports[`Thrift Parser > view token 80`] = `
"[line=12] [type=23] [channel=0] [tokenIndex=79] [startIndex=347] [stopIndex=347]
)"
`;

exports[`Thrift Parser > view token 81`] = `
"[line=12] [type=50] [channel=0] [tokenIndex=80] [startIndex=348] [stopIndex=348]
,"
`;

exports[`Thrift Parser > view token 82`] = `
"[line=12] [type=51] [channel=1] [tokenIndex=81] [startIndex=349] [stopIndex=354]


    "
`;

exports[`Thrift Parser > view token 83`] = `
"[line=14] [type=53] [channel=2] [tokenIndex=82] [startIndex=355] [stopIndex=371]
/* Foo Comment */"
`;

exports[`Thrift Parser > view token 84`] = `
"[line=14] [type=51] [channel=1] [tokenIndex=83] [startIndex=372] [stopIndex=376]

    "
`;

exports[`Thrift Parser > view token 85`] = `
"[line=15] [type=26] [channel=0] [tokenIndex=84] [startIndex=377] [stopIndex=380]
void"
`;

exports[`Thrift Parser > view token 86`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=85] [startIndex=381] [stopIndex=381]
 "
`;

exports[`Thrift Parser > view token 87`] = `
"[line=15] [type=49] [channel=0] [tokenIndex=86] [startIndex=382] [stopIndex=384]
Foo"
`;

exports[`Thrift Parser > view token 88`] = `
"[line=15] [type=22] [channel=0] [tokenIndex=87] [startIndex=385] [stopIndex=385]
("
`;

exports[`Thrift Parser > view token 89`] = `
"[line=15] [type=23] [channel=0] [tokenIndex=88] [startIndex=386] [stopIndex=386]
)"
`;

exports[`Thrift Parser > view token 90`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=89] [startIndex=387] [stopIndex=387]
 "
`;

exports[`Thrift Parser > view token 91`] = `
"[line=15] [type=22] [channel=0] [tokenIndex=90] [startIndex=388] [stopIndex=388]
("
`;

exports[`Thrift Parser > view token 92`] = `
"[line=15] [type=49] [channel=0] [tokenIndex=91] [startIndex=389] [stopIndex=396]
api.post"
`;

exports[`Thrift Parser > view token 93`] = `
"[line=15] [type=8] [channel=0] [tokenIndex=92] [startIndex=397] [stopIndex=397]
="
`;

exports[`Thrift Parser > view token 94`] = `
"[line=15] [type=48] [channel=0] [tokenIndex=93] [startIndex=398] [stopIndex=403]
\\"/Foo\\""
`;

exports[`Thrift Parser > view token 95`] = `
"[line=15] [type=50] [channel=0] [tokenIndex=94] [startIndex=404] [stopIndex=404]
,"
`;

exports[`Thrift Parser > view token 96`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=95] [startIndex=405] [stopIndex=405]
 "
`;

exports[`Thrift Parser > view token 97`] = `
"[line=15] [type=49] [channel=0] [tokenIndex=96] [startIndex=406] [stopIndex=420]
api.data_policy"
`;

exports[`Thrift Parser > view token 98`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=97] [startIndex=421] [stopIndex=421]
 "
`;

exports[`Thrift Parser > view token 99`] = `
"[line=15] [type=8] [channel=0] [tokenIndex=98] [startIndex=422] [stopIndex=422]
="
`;

exports[`Thrift Parser > view token 100`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=99] [startIndex=423] [stopIndex=423]
 "
`;

exports[`Thrift Parser > view token 101`] = `
"[line=15] [type=48] [channel=0] [tokenIndex=100] [startIndex=424] [stopIndex=431]
\\"ENABLE\\""
`;

exports[`Thrift Parser > view token 102`] = `
"[line=15] [type=50] [channel=0] [tokenIndex=101] [startIndex=432] [stopIndex=432]
,"
`;

exports[`Thrift Parser > view token 103`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=102] [startIndex=433] [stopIndex=433]
 "
`;

exports[`Thrift Parser > view token 104`] = `
"[line=15] [type=49] [channel=0] [tokenIndex=103] [startIndex=434] [stopIndex=447]
api.serializer"
`;

exports[`Thrift Parser > view token 105`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=104] [startIndex=448] [stopIndex=448]
 "
`;

exports[`Thrift Parser > view token 106`] = `
"[line=15] [type=8] [channel=0] [tokenIndex=105] [startIndex=449] [stopIndex=449]
="
`;

exports[`Thrift Parser > view token 107`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=106] [startIndex=450] [stopIndex=450]
 "
`;

exports[`Thrift Parser > view token 108`] = `
"[line=15] [type=48] [channel=0] [tokenIndex=107] [startIndex=451] [stopIndex=456]
\\"json\\""
`;

exports[`Thrift Parser > view token 109`] = `
"[line=15] [type=50] [channel=0] [tokenIndex=108] [startIndex=457] [stopIndex=457]
,"
`;

exports[`Thrift Parser > view token 110`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=109] [startIndex=458] [stopIndex=458]
 "
`;

exports[`Thrift Parser > view token 111`] = `
"[line=15] [type=49] [channel=0] [tokenIndex=110] [startIndex=459] [stopIndex=477]
api.resp_serializer"
`;

exports[`Thrift Parser > view token 112`] = `
"[line=15] [type=8] [channel=0] [tokenIndex=111] [startIndex=478] [stopIndex=478]
="
`;

exports[`Thrift Parser > view token 113`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=112] [startIndex=479] [stopIndex=479]
 "
`;

exports[`Thrift Parser > view token 114`] = `
"[line=15] [type=48] [channel=0] [tokenIndex=113] [startIndex=480] [stopIndex=485]
\\"json\\""
`;

exports[`Thrift Parser > view token 115`] = `
"[line=15] [type=23] [channel=0] [tokenIndex=114] [startIndex=486] [stopIndex=486]
)"
`;

exports[`Thrift Parser > view token 116`] = `
"[line=15] [type=50] [channel=0] [tokenIndex=115] [startIndex=487] [stopIndex=487]
,"
`;

exports[`Thrift Parser > view token 117`] = `
"[line=15] [type=51] [channel=1] [tokenIndex=116] [startIndex=488] [stopIndex=493]


    "
`;

exports[`Thrift Parser > view token 118`] = `
"[line=17] [type=52] [channel=2] [tokenIndex=117] [startIndex=494] [stopIndex=507]
// Deprecated
"
`;

exports[`Thrift Parser > view token 119`] = `
"[line=18] [type=51] [channel=1] [tokenIndex=118] [startIndex=508] [stopIndex=511]
    "
`;

exports[`Thrift Parser > view token 120`] = `
"[line=18] [type=26] [channel=0] [tokenIndex=119] [startIndex=512] [stopIndex=515]
void"
`;

exports[`Thrift Parser > view token 121`] = `
"[line=18] [type=51] [channel=1] [tokenIndex=120] [startIndex=516] [stopIndex=516]
 "
`;

exports[`Thrift Parser > view token 122`] = `
"[line=18] [type=49] [channel=0] [tokenIndex=121] [startIndex=517] [stopIndex=526]
Deprecated"
`;

exports[`Thrift Parser > view token 123`] = `
"[line=18] [type=22] [channel=0] [tokenIndex=122] [startIndex=527] [stopIndex=527]
("
`;

exports[`Thrift Parser > view token 124`] = `
"[line=18] [type=23] [channel=0] [tokenIndex=123] [startIndex=528] [stopIndex=528]
)"
`;

exports[`Thrift Parser > view token 125`] = `
"[line=18] [type=51] [channel=1] [tokenIndex=124] [startIndex=529] [stopIndex=529]
 "
`;

exports[`Thrift Parser > view token 126`] = `
"[line=18] [type=22] [channel=0] [tokenIndex=125] [startIndex=530] [stopIndex=530]
("
`;

exports[`Thrift Parser > view token 127`] = `
"[line=18] [type=49] [channel=0] [tokenIndex=126] [startIndex=531] [stopIndex=540]
deprecated"
`;

exports[`Thrift Parser > view token 128`] = `
"[line=18] [type=23] [channel=0] [tokenIndex=127] [startIndex=541] [stopIndex=541]
)"
`;

exports[`Thrift Parser > view token 129`] = `
"[line=18] [type=50] [channel=0] [tokenIndex=128] [startIndex=542] [stopIndex=542]
,"
`;

exports[`Thrift Parser > view token 130`] = `
"[line=18] [type=51] [channel=1] [tokenIndex=129] [startIndex=543] [stopIndex=543]

"
`;

exports[`Thrift Parser > view token 131`] = `
"[line=19] [type=12] [channel=0] [tokenIndex=130] [startIndex=544] [stopIndex=544]
}"
`;

exports[`Thrift Parser > view token 132`] = `
"[line=19] [type=51] [channel=1] [tokenIndex=131] [startIndex=545] [stopIndex=545]

"
`;

exports[`Thrift Parser > view token 133`] = `
"[line=20] [type=-1] [channel=0] [tokenIndex=132] [startIndex=546] [stopIndex=545]
<EOF>"
`;
